src/engine/sync/enginesync.cpp

```diff
@@ -32,17 +38,76 @@ EngineSync::EngineSync(UserSettingsPointer pConfig)
 EngineSync::~EngineSync() {
 }
 
+Syncable* EngineSync::pickMaster(Syncable* enabling_syncable) {
+    // If there is an explicit master, and it is playing, keep it.
+    if (m_pMasterSyncable && m_pMasterSyncable->getSyncMode() == SYNC_MASTER_EXPLICIT && m_pMasterSyncable->isPlaying()) {
+        return m_pMasterSyncable;
+    }
+
+    static std::vector<Syncable*> stopped_sync_decks;
```

Comment:
I don't understand why static variables are used here.

Reply:
did you read my replies and linked discussions?

Reply:
Code should be comprehensible without any additional context that is not accessible outside of this scope.

Reply:
I'm talking about the explanations for why it's not necessary to avoid 100% of allocations is not productive.  My concern is that a solution to preallocate or do allocation in another thread is overcomplicated and unnecessary

Reply:
Why don't you use ordinary member variables and do the allocations during construction? The constructor doesn't seem to be invoked in time critical sections.

I still don't get it why we need to use static variables hidden inside a non-static member function that are shared among *all* instances of this class. This design is not sound.

Reply:
member variables for temporary workers in a rarely-executed function is not great design either.  how about a preallocated std::array?